Jenkinsfile

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env groovy
+
+node {
+  def govuk = load '/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy'
+
+  try {
+    stage("Checkout") {
+      checkout scm
+    }
+
+    stage("Cleanup") {
+      govuk.cleanupGit()
+      govuk.mergeMasterBranch()
+    }
+
+    stage("Build") {
+      sh "${WORKSPACE}/jenkins.sh"
+    }
+
+    stage("Publish gem") {
+      if (env.BRANCH_NAME == 'master') {
+        sh "bundle exec rake publish_gem --trace"
```

Comment:
This won't work until https://github.com/alphagov/govuk-puppet/pull/5325 is merged and deployed. Doesn't necessarily mean this branch can't be merged though, just something to be aware of.